File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if 0 == len(s):
            return False

        lows = s.lower()
        begin = 0
        end = len(lows) - 1
        while begin <= end:
            if not lows[begin].isalnum():
                begin += 1
                continue

            if not lows[end].isalnum():
                end -= 1
                continue

            if lows[begin] == lows[end]:
                begin += 1
                end -= 1
            else:
                return False
        
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^0-9A-Za-z]", "").toLowerCase();
        // System.out.println(s);
        for (int begin = 0, end = s.length() - 1; begin < end; begin++, end--) {
            if (s.charAt(begin) != s.charAt(end)) {
                return false;
            }
        }

        return true;
    }
}